content/blog/exposing-an-application-using-ingress-and-tls-termination-on-kubernetes-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,528 @@
+deployment.apps/nginx-blue    1/1     1            1           15s
+deployment.apps/nginx-green   1/1     1            1           24s
+deployment.apps/nginx-main    1/1     1            1           32s
+
+NAME                                    DESIRED   CURRENT   READY   AGE
+replicaset.apps/nginx-blue-78647f4c4b   1         1         1       15s
+replicaset.apps/nginx-green-8956bbd9f   1         1         1       24s
+replicaset.apps/nginx-main-64bfd77895   1         1         1       32s
+```
+
+Three Nginx services, *nginx-main*, *nginx-blue* and *nginx-green*, are deployed as the *ClusterIP* type. They provide internal connectivity and can solely be accessed from within the cluster.
+
+T﻿ype the following commend to check that all the application service endpoints have been populated:
+
+```shell
+$ kubectl get endpoints -n nginx-apps
+NAME          ENDPOINTS        AGE
+nginx-blue    10.192.3.78:80    1m
+nginx-green   10.192.4.45:80    1m
+nginx-main    10.192.4.44:80    1m
+```
+
+### Set up Ingress TLS
+
+The Ingress resource with TLS has to be created. Here is the sample Ingress TLS resource *ingress-host-based-selfsigned.yaml*, available from the GitHub repo [ingress-demo](https://github.com/GuopingJia/ingress-demo):
+
+```shell
+$ cat ingress-host-based-selfsigned.yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-host-based-selfsigned
+  annotations:
+    ingress.kubernetes.io/ssl-redirect: "true"
+    cert-manager.io/issuer: "nginx-selfsigned-issuer"
+spec:
+  ingressClassName: nginx
+  tls:
+  - hosts:
+    - nginx.example.com
+    secretName: cfe-tls-key-pair
+  rules:
+  - host: nginx.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: nginx-main
+            port:
+              number: 80
+  - host: blue.nginx.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: nginx-blue
+            port:
+              number: 80
+  - host: green.nginx.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: nginx-green
+            port:
+              number: 80
+```
+
+In the above sample YAML manifest file, there is the *'tls'* block that contains the hostname *'nginx.example.com'* and the secret *cfe-tls-key-pair* created in the certification step. There is also the *'rules'* block in which a list of routing rules is defined per host, e.g., host *nginx.example.com* will be routed to the application service *nginx-main* in the backend.  
+
+T﻿ype the following command to deploy the Ingress resource to the namespace *nginx-apps*:
+
+```shell
+$ kubectl apply -f ingress-host-based-selfsigned.yaml -n nginx-apps
+ingress.networking.k8s.io/ingress-host-based-selfsigned created
+```
+
+Check the details of the *TLS* and *Rules* settings by t﻿yping below command:
```

Comment:
The last part of this sentence needs to be modified. Try this:
"Check the details of the *TLS* and *Rules* settings by typing the command shown below:"